### DISCUSSION TITLE
Debug high memory usage

### DISCUSSION THREAD

**Discussion by jpellegrini** [egallesio/STklos#189] [Q&A]
Hi @egallesio !
I was working on SRFI-194,  and its test suite always eat up all available memory (I have 32Gb RAM!)
I was wondering if there is a way to debug memory allocation and tell, at least, what function is allocating more memory, or something similar.

**Comment by egallesio** [comment]
>  was wondering if there is a way to debug memory allocation and tell, at least, what function is allocating more memory, or something similar.

There are several compilation flags for the GC (see in gc/doc) . I always found that they  are more useful for debugging the gc library than your program. A simple way to debug I have used is tio change  `STk_must_malloc` (it is a macro by something like

```c
#define STk_must_malloc(size)           debug_gc_malloc(size, __FILE__, __LINE__)
```
and implement the wrapper  `debug_gc_malloc` around GC_MALLOC 
This is basic, but perhaps it could suffice.
